packages/core/configuration/configurationSchema.ts

```diff
@@ -169,26 +123,33 @@ export function ConfigurationSchema(
     },
   }
   Object.entries(schemaDefinition).forEach(([slotName, slotDefinition]) => {
-    if (isConfigurationSchemaType(slotDefinition)) {
-      // this is a sub-configuration
+    if (
+      (isType(slotDefinition) && isLateType(slotDefinition)) ||
+      isConfigurationSchemaType(slotDefinition)
+    ) {
+      // this is either an MST late() type (which we assume to be a sub-configuration),
+      // or an actual sub-configuration
       modelDefinition[slotName] = slotDefinition
+    } else if (
+      typeof slotDefinition === 'string' ||
+      typeof slotDefinition === 'number'
+    ) {
+      volatileConstants[slotName] = slotDefinition
     } else if (typeof slotDefinition === 'object') {
       // this is a slot definition
-      if (!slotDefinition.type) {
+      if (!('type' in slotDefinition) || !slotDefinition.type) {
```

Comment:
any reason for this extra check guard?

Reply:
not really, i'll remove it

Reply:
i recall 'type' in slotDefinition is one way to avoid directly checking slotDefinition.type which typescript complains of, something I discovered with peter recently

Reply:
so if it helps in that way, it's fine